### PR TITLE
build: Error out properly when unsupported arch is given.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Release build with unsupported arch selected should error out.

## Motivation and Context
Release build cleanup.

## How Has This Been Tested?
Manually with `make release`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by donatello — APPROVED** [review]

File: buildscripts/build.sh [hunk 1]
```diff
@@ -90,9 +90,18 @@ main() {
             go_build ${each_osarch}
         done
     else
-        for each_osarch in $(echo $chosen_osarch | sed 's/,/ /g'); do
-            go_build ${each_osarch}
+        local found=0
+        for each_osarch in ${SUPPORTED_OSARCH}; do
+            if [ "$chosen_osarch" = "$each_osarch" ]; then
```
Comment (nitisht) [inline buildscripts/build.sh:95]: Probably more readable with `==` instead of `=`

**Review by nitisht — CHANGES_REQUESTED** [review]

File: buildscripts/build.sh (hunk 1 above)
Comment (harshavardhana) [reply buildscripts/build.sh:95]: No @NitishT - test it locally `=` is also same as `==`  in bash.

Comment (nitisht) [reply buildscripts/build.sh:95]: I see. Let me build locally.

**Review by nitisht — APPROVED** [review]
LGTM.

**Pull request merged**